liquid-rust-typeck/src/checker.rs

```diff
@@ -84,6 +84,16 @@ pub struct Check<'a> {
     bb_envs: FxHashMap<BasicBlock, BasicBlockEnv>,
     kvars: &'a mut KVarStore,
 }
+/// A 'Guard' describes extra "control" information that holds at the start
+/// of the successor basic block
+enum Guard {
```

Comment:
hmm I'm ambivalent about this, it is a good abstraction if we ever need to track variants, but it makes the code a bit complicated in some parts. I'd personally remove it, but no strong opinions.